lib/mars/aggregator.rb

```diff
@@ -2,16 +2,15 @@
 
 module Mars
   class Aggregator < Runnable
-    attr_reader :name
+    attr_reader :name, :operation
 
-    def initialize(name = "Aggregator")
+    def initialize(name = "Aggregator", operation: nil)
       @name = name
+      @operation = operation || ->(inputs) { inputs.join("\n") }
     end
 
     def run(inputs)
-      return yield if block_given?
-
-      inputs.join("\n")
+      operation.call(inputs)
```

Comment:
why do we need this operation? I'm a bit worried about allowing random Ruby code because it may affect out ability to accurately describe workflows as diagrams right?

what I mean is that I could have an operation be `->(inputs) { raise 'error' }` and then we don't have any way to detect that during the diagram rendering